public/app/core/components/DeleteButton/DeleteButton.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { Component } from 'react';
+
+export default class DeleteButton extends Component<any, any> {
+  state = {
```

Comment:
It's a bit unusual to use render-specifics like CSS class names as the state values. I would recommend against it, because it's much easier to reason later in the render function on what's going on. Try modelling the state with true/false values instead.

The state is essentially binary: delete vs showing the confirm button, so one state value (e.g. `confirmed: false`) could suffice.